ksql-engine/src/main/java/io/confluent/ksql/function/UdfArgCoercer.java

```diff
@@ -16,57 +16,22 @@
 package io.confluent.ksql.function;
 
 import com.google.common.primitives.Primitives;
-import com.squareup.javapoet.CodeBlock;
 import java.lang.reflect.Array;
-import java.lang.reflect.Method;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
-@SuppressWarnings("WeakerAccess")
-public final class UdfTemplate {
+final class UdfArgCoercer {
 
-  private static final Logger LOG = LoggerFactory.getLogger(UdfTemplate.class);
-
-  private UdfTemplate() { }
-
-  static String generateCode(final Method method, final String obj) {
-    final Class<?>[] params = method.getParameterTypes();
-
-    final CodeBlock.Builder code = CodeBlock.builder();
-    for (int idx = 0; idx < params.length; idx++) {
-      final Class<?> param = params[idx];
-      code.addStatement("$T arg$L = ($T) $T.coerce(args, $T.class, $L)",
-                        Primitives.wrap(param),
-                        idx,
-                        Primitives.wrap(param),
-                        UdfTemplate.class,
-                        param,
-                        idx);
-    }
-
-    final String args = IntStream.range(0, params.length)
-        .mapToObj(i -> "arg" + i)
-        .collect(Collectors.joining(", "));
-
-    code.addStatement("return (($L) $L).$L($L)",
-        method.getDeclaringClass().getCanonicalName(), obj, method.getName(), args);
-
-    final String codeString = code.build().toString();
-    LOG.trace("Generated code:\n" + codeString);
-    return codeString;
+  private UdfArgCoercer() {
   }
 
-  public static <T> T coerce(
+  static <T> T coerceUdfArgs(
```

Comment:
Isn't this only called from tests? It was in the old code, so maybe consider inlining...